core/src/calleeFunctions/index.ts

```diff
@@ -23,27 +23,97 @@ const allCalleeFunctions: Record<CalleeNames, CalleeFunctions> = {
 export const getCalleeData = async function (
     network: string,
     collateralType: string,
+    marketId: string,
     profitAddress: string
 ): Promise<string> {
     const collateral = getCollateralConfigByType(collateralType);
-    const calleeFuctions = allCalleeFunctions[collateral.exchange.callee];
-    if (!calleeFuctions) {
+    const marketData = collateral.exchanges[marketId];
+    if (!marketData || !marketData.callee || !allCalleeFunctions[marketData.callee]) {
         throw new Error(`Unsupported collateral type "${collateralType}"`);
     }
-    return await calleeFuctions.getCalleeData(network, collateral, profitAddress);
+    return await allCalleeFunctions[marketData.callee].getCalleeData(network, collateral, marketId, profitAddress);
 };
 
-const _getMarketPrice = async function (
+export const getMarketData = async function (
     network: string,
     collateralSymbol: string,
     amount: BigNumber = new BigNumber('1')
-): Promise<BigNumber> {
+): Promise<Record<string, MarketData>> {
     const collateral = getCollateralConfigBySymbol(collateralSymbol);
-    const calleeFuctions = allCalleeFunctions[collateral.exchange.callee];
-    if (!calleeFuctions) {
+    let isCollateralSupported = false;
+    for (const market of Object.values(collateral.exchanges)) {
+        if (market.callee && allCalleeFunctions[market.callee]) {
+            isCollateralSupported = true;
+            break;
+        }
+    }
+    if (!isCollateralSupported) {
         throw new Error(`Unsupported collateral symbol "${collateralSymbol}"`);
     }
```

Comment:
Nit: most of the time the usage of let can be avoided by better structured code
```suggestion
    const isCollateralSupported = Object.values(collateral.exchanges).some(market => market.callee && allCalleeFunctions[market.callee])
    if (!isCollateralSupported) {
        throw new Error(`Unsupported collateral symbol "${collateralSymbol}"`);
    }
```